app/controllers/hosts_controller.rb

```diff
@@ -199,7 +199,7 @@ def externalNodes
     begin
       respond_to do |format|
         # don't break lines in yaml to support Ruby < 1.9.3
-        host_info_yaml = @host.info.to_yaml(:line_width => -1)
+        host_info_yaml = JSON.parse(@host.info.to_json).to_yaml(:line_width => -1)
```

Comment:
Using `#deep_to_hash`[1] should be more effective than serializing and then deserializing the data and yield the same results. Is there a reason for not using it?

[1] - https://github.com/theforeman/foreman/blob/develop/config/initializers/active_support_extensions.rb#L5

Reply:
Not really, I will use that

Reply:
Actually.. yeah, the issue is that the ENC result is not a ActiveSupport hash. It's only the hashes inside arrays in a YAML smart class parameter:

```ruby
"classes"=>{"dhcp"=>{"bootfiles"=>[{"name"=>"foo", "mount_point"=>"/bar"}, {"name"=>"john", "mount_point"=>"/doe"}]}}, "environment"=>"production"}

irb(main):009:0> a['parameters'].class
=> Hash
irb(main):010:0> a['classes'].class
=> Hash
irb(main):011:0> a['classes']['dhcp'].class
=> Hash
irb(main):012:0> a['classes']['dhcp']['bootfiles'].class
=> Array
irb(main):013:0> a['classes']['dhcp']['bootfiles'].first.class
=> ActiveSupport::HashWithIndifferentAccess
```

Reply:
Ah, I see. I wrongly assumed the whole thing would be IndifferentHash. Nevermind then

Reply:
What about `Hash#deep_stringify_keys`?:
```ruby
[17] pry(main)> a = {'a' => [{'b' => 'c'}.with_indifferent_access]}
=> {"a"=>[{"b"=>"c"}]}
[18] pry(main)> a.to_yaml
=> "---\na:\n- !ruby/hash:ActiveSupport::HashWithIndifferentAccess\n  b: c\n"
[20] pry(main)> a.deep_stringify_keys.to_yaml
=> "---\na:\n- b: c\n"